frontend/src/store/middlewares/socket/socket.middleware.ts

```diff
@@ -0,0 +1,37 @@
+import { Middleware } from '@reduxjs/toolkit';
+import { SocketEvent } from 'common/enums/enums';
+import {
+  AppDispatch,
+  ChatMessageGetAllItemResponseDto,
+} from 'common/types/types';
+import { socket as socketService } from 'services/services';
+import { chatsActions } from 'store/actions';
+
+type SocketMiddlewareParams = {
+  dispatch: AppDispatch;
+};
+
+const socket: Middleware =
+  ({ dispatch }: SocketMiddlewareParams) =>
+  (next) =>
+  (action): void => {
+    const socketInstance = socketService.socket;
+
+    const newMessageListener = (
+      chatMessage: ChatMessageGetAllItemResponseDto,
+    ): void => {
+      dispatch(chatsActions.getNewMessage(chatMessage));
+    };
+
+    if (chatsActions.getMessages.pending.match(action)) {
+      socketInstance.off(SocketEvent.MESSAGE);
+    }
+
+    if (chatsActions.getMessages.fulfilled.match(action)) {
+      socketInstance.on(SocketEvent.MESSAGE, newMessageListener);
+    }
```

Comment:
Also pay attention that CHAT_ADD_MESSAGE event should be sent only for receiver
```suggestion
    chatSocket.on(SocketEvent.CHAT_ADD_MESSAGE, (message: ChatMessageGetAllItemResponseDto): void => {
      dispatch(chatsActions.addMessage(message));
    });
    
    if (chatsActions.joinRoom.match(action)) {
      chatSocket.emit(SocketEvent.CHAT_JOIN_ROOM, action.payload);
    }
    
   if (chatsActions.leaveRoom.match(action)) {
      chatSocket.emit(SocketEvent.CHAT_LEAVE_ROOM, action.payload);
    }

    if (chatsActions.createMessage.fulfilled.match(action)) {
      chatSocket.emit(SocketEvent.CHAT_CREATE_MESSAGE, action.payload);
    }
```